src/unix/tcp.c

```diff
@@ -156,7 +156,9 @@ int uv__tcp_connect(uv_connect_t* req,
   if (err)
     return err;
 
-  handle->delayed_error = 0;
+  if (handle->delayed_error) {
+    return handle->delayed_error;
+  }
```

Comment:
I would move this check up before the call to `maybe_new_socket()`.  Also, can you drop the braces?